src/liballoc/string.rs

```diff
@@ -677,13 +696,15 @@ impl String {
     ///
     /// unsafe {
     ///     let s = String::from("hello");
-    ///     let ptr = s.as_ptr();
+    ///
+    ///     let mut s = mem::ManuallyDrop::new(s);
```

Comment:
I know the old code had no comments either, but I feel like at least this line warrants a comment (similar to the other examples you changed)